pagerduty/service.go

```diff
@@ -51,8 +51,8 @@ type AlertGroupingParameters struct {
 
 // AutoPauseNotificationsParameters defines how alerts on this service are automatically suspended for a period of time before triggering, when identified as likely being transient.
 type AutoPauseNotificationsParameters struct {
-	Enabled bool `json:enabled,omitempty`
-	Timeout int  `json:timeout,omitempty`
+	Enabled bool `json:"enabled"`
+	Timeout int  `json:"timeout"`
```

Comment:
Turns out the API returns `null` for `timeout` if `enabled` is set to false:

```
---[ REQUEST ]---------------------------------------
PUT /services/PDKFHYX HTTP/1.1
Host: api.pagerduty.com
User-Agent: heimweh/go-pagerduty(terraform)
Content-Length: 385
Accept: application/vnd.pagerduty+json;version=2
Authorization: Token token=******
Content-Type: application/json
Accept-Encoding: gzip

{
 "service": {
  "acknowledgement_timeout": 1800,
  "alert_creation": "create_alerts_and_incidents",
  "alert_grouping": null,
  "auto_pause_notifications_parameters": {
   "enabled": false,
   "timeout": 300
  },
  "auto_resolve_timeout": 1800,
  "description": "foo",
  "escalation_policy": {
   "id": "PPF7AEC",
   "type": "escalation_policy_reference"
  },
  "incident_urgency_rule": {
   "type": "constant",
   "urgency": "high"
  },
  "name": "tf-43e4n"
 }
}

-----------------------------------------------------
2022/06/10 15:06:33 [DEBUG] PagerDuty API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/2.0 200 OK
Access-Control-Allow-Headers: Authorization, Content-Type, AuthorizationOauth, X-EARLY-ACCESS
Access-Control-Allow-Methods: GET, POST, PUT, DELETE, OPTIONS
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers:
Access-Control-Max-Age: 1728000
Cache-Control: max-age=0, private, must-revalidate
Content-Type: application/json
Date: Fri, 10 Jun 2022 13:06:33 GMT
Etag: W/"45b125457a612501fdfc61e82a2213e0"
Feature-Policy: accelerometer 'none'; camera 'none'; geolocation 'none'; gyroscope 'none'; magnetometer 'none'; microphone 'none'; payment 'none'; usb 'none'
Referrer-Policy: strict-origin-when-cross-origin
Referrer-Policy: strict-origin-when-cross-origin
Server: nginx
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Accept-Encoding
X-Content-Type-Options: nosniff
X-Request-Id: 218f2e5d06395dc71236b61febd23c66
X-Xss-Protection: 1; mode=block

{
 "service": {
  "id": "PDKFHYX",
  "name": "tf-43e4n",
  "description": "foo",
  "created_at": "2022-06-10T15:06:29+02:00",
  "updated_at": "2022-06-10T15:06:33+02:00",
  "status": "active",
  "teams": [],
  "alert_creation": "create_alerts_and_incidents",
  "addons": [],
  "scheduled_actions": [],
  "support_hours": null,
  "last_incident_timestamp": null,
  "escalation_policy": {
   "id": "PPF7AEC",
   "type": "escalation_policy_reference",
   "summary": "tf-xgyxe",
   "self": "https://api.pagerduty.com/escalation_policies/PPF7AEC",
   "html_url": "https://dev-claranet.pagerduty.com/escalation_policies/PPF7AEC"
  },
  "incident_urgency_rule": {
   "type": "constant",
   "urgency": "high"
  },
  "acknowledgement_timeout": 1800,
  "auto_resolve_timeout": 1800,
  "auto_pause_notifications_parameters": {
   "enabled": false,
   "timeout": null
  },
  "alert_grouping": null,
  "alert_grouping_timeout": null,
  "alert_grouping_parameters": {
   "type": null,
   "config": null
  },
  "integrations": [],
  "response_play": null,
  "type": "service",
  "summary": "tf-43e4n",
  "self": "https://api.pagerduty.com/services/PDKFHYX",
  "html_url": "https://dev-claranet.pagerduty.com/service-directory/PDKFHYX"
 }
}
-----------------------------------------------------
```

`timeout` probably needs to be an int pointer instead of an int.